File: 2025/1-day/task1.ts
import * as fs from "fs";

const dataLocation = "2025/1-day/data.txt";

let value = 50;
let numberOfZeroes = 0;

fs.readFile(dataLocation, (err, data) => {
  if (err) throw err;
  const start = performance.now();
  const input = data.toString().split("\r\n");
  input.forEach((val) => {
    const tempVal = Number(val.slice(1));
    for (let index = 0; index < tempVal; index++) {
      if (val.slice(0, 1) === "L") {
        if (value === 0) {
          value = 99;
          continue;
        }
        value--;
      } else {
        if (value === 99) {
          value = 0;
          continue;
        }
        value++;
      }
    }
    if (value === 0) {
      numberOfZeroes++;
    }
  });
  const end = performance.now();
  const duration = end - start;
  console.log(`Time: ${duration.toFixed(2)}ms`);
  console.log(numberOfZeroes);
});

File: 2025/1-day/task1.py
import time

dataLocation = "2025/1-day/data.txt"
value = 50
numberOfZeroes = 0

content = open(dataLocation)

start = time.perf_counter()
splitRecords = content.read().split()

actions = [(a[0], int(a[1:])) for a in content.read().split()]

for direction, steps in actions:
    if direction == "L":
        value = (value - steps) % 100
    else:
        value = (value + steps) % 100
        
    if value == 0:
        numberOfZeroes += 1
    
print(numberOfZeroes)
end = time.perf_counter()
print(f"{(end - start) * 1000:.2f} ms")